cdk/domino_cdk/provisioners/eks/eks_cluster.py

```diff
@@ -88,6 +88,28 @@ def provision(
             },
         )
 
+        params = {
+            "resourceArn": f"arn:aws:eks:{self.scope.region}:{self.scope.account}:cluster/{cluster.cluster_name}",
+            "tags": {
+                "domino-deploy-id": self.scope.name,
```

Comment:
OK, question: 

There we have 
```
tags:
  domino-infrastructure: "true"
```

but we do in the code:

`cdk.Tags.of(self).add("domino-deploy-id", self.name)`

Please advise

Reply:
Right below that it iterates the user-provided global tags as well. Since we didn't reuse the variable anywhere else, it was fine. I think we need to switch to injecting our hardcoded deploy id tag into the cfg before the iteration in `cdk/domino_cdk/domino_stack.py`:

```
        self.cfg.tags["domino-deploy-id"] = self.name
        for k, v in self.cfg.tags.items():
            cdk.Tags.of(self).add(str(k), str(v))
```

Then we can use `self.cfg.tags` in the EKS cluster and future places where we need to manually tag things to address cdk/cf tagging deficiencies.

Reply:
I understand this comment as:
1. This was the only place, now we have many, let's make it good.
2. Instead of this separate line, let's force-add "domino-deploy-id" tag to the tags in the config object when reading config and then do generic tag iteration everywhere we use tags.

I'll do it.

Probably difference with what you are saying: 
* I think we add the mandatory tag when reading the tags from the config file
* it looks like you are saying something different like adding the mandatory tag right before first usage in one of the classes using the tags. 
 
I think we should not do it here because then the config becomes not idempotent on the order of the classes called. Tomorrow we could add something to pass cfg to and call it before DominoStack.__init__() the there tags will be partial. 

The config reader should return complete and immutable config and we should avoid side-effects of mutating config during processing.

Please comment if you have any concerns.

Reply:
I think there are two things that make sense.

* Either we do it first thing in `DominoStack.init()` as suggested, in which case it's just a mandatory hardcoded global tag
* Or we put it into the default template, ie: https://github.com/dominodatalab/cdk-cf-eks/blob/9ebcafd778871d05f22d0928ddb87fc799e62576/cdk/domino_cdk/config/template.py#L154
  * In this version, it's just a suggestion.

I don't have a strong preference, but there is a small operational difference between the two.

There could be another version of the first option where we make a function that returns the combined hardcoded tag + tag object, and everything that access global tags is wrapped through that function. Then we're not mutating the config object ever.

Reply:
I do not think the third option is a variant of the first.

First option: we pass config object to stack constructor and it alters the config.

Second option: we somehow alter the config between reading it from the file and returning it. Yes, we can do it with the default template by adding default values.

Hmm. I would actually prefer to give an error if the user tries to override domino-deployment-id. I do not think it is that Linus' "bullet-foot delivery" case, therefore we should not allow the customers to override domino-deployment-id to be different from the stack name.

Third option: do not alter a config object in memory but add a getter that will alter a copy of a part of the config in between reading it from the memory object and returning it to a caller

Reply:
Wait a sec. We were talking that stack can be reused for different deployments.

Would deployer re-tag with "deployment id" tag?

Reply:
I don't think we're expecting to use this with multi-tenant deployments as-is anyway. There will need to be thought outside the scope of just the tagging to how we handle the s3 buckets and provisioning that I'm sure our tagging strategy would have to come out of.

Reply:
OK, then I'm not worrying about this for now.